libs/wallet/src/vega-transaction-dialog/vega-transaction-dialog.tsx

```diff
@@ -0,0 +1,137 @@
+import { OrderStatus } from '@vegaprotocol/types';
+import { Dialog, Intent } from '@vegaprotocol/ui-toolkit';
+import { useEffect } from 'react';
+import type { VegaTxState } from '../use-vega-transaction';
+import { VegaTxStatus } from '../use-vega-transaction';
+import type { Order } from '../vega-order-transaction-dialog';
+import { VegaOrderTransactionDialog } from '../vega-order-transaction-dialog';
+
+export interface VegaTransactionDialogProps {
+  orderDialogOpen: boolean;
+  setOrderDialogOpen: (isOpen: boolean) => void;
+  finalizedOrder: Order | null;
+  transaction: VegaTxState;
+  reset: () => void;
+  type: VegaOrderTransactionType;
+}
+
+export enum VegaOrderTransactionType {
+  SUBMIT = 'submit',
+  CANCEL = 'cancel',
+  EDIT = 'edit',
```

Comment:
This dialog will change in the next PR https://github.com/vegaprotocol/frontend-monorepo/issues/470